### ISSUE TITLE
Report exceptions to master

### ISSUE THREAD

**Issue by valpackett** [LearnBoost/cluster#41]
I haven't started using cluster yet, but I saw this in worker.js:

```
process.on('uncaughtException', function(err){
    console.error(err.stack || err.message);
    self.destroy();
});
```

This should be optional because (if I'm not wrong... oh, asynchronous stuff), self.destroy() might be called before [exceptional-node](https://github.com/contrast/exceptional-node)'s sending call.

**Comment by tj** [comment]
I have an issue open for better plugin / worker integration. Currently the plugins operate within master only. one possibility is that on uncaughtException we could just serialize as JSON, send it to master, then die

**Comment by tj** [comment]
(changing the title to be more generic)

**Comment by valpackett** [comment]
So right now I have to use try/catch to get my exceptions sent to Exceptional?

BTW, how about a custom callback insertion into workers?

**Comment by tj** [comment]
Yeah I would like to have plugins operate in workers to, in which case you could just catch that and do whatever you want, but it would just get a little confusing for people developing plugins haha. I will report back to master for now

**Comment by tj** [comment]
Added &quot;worker exception&quot; event. Closed by 5bb58b4d7764d34e94c437c6c263538fb42e42c4

**Comment by valpackett** [comment]
Wow, that's fast! You rock.
